addons/mail/static/tests/translation/translation_test.js

```diff
@@ -0,0 +1,44 @@
+/* @odoo-module */
+
+import { start } from "@mail/../tests/helpers/test_utils";
+import { click, contains } from "@web/../tests/utils";
+import { startServer } from "@bus/../tests/helpers/mock_python_environment";
+
+QUnit.module("Google Cloud Translation");
+
+QUnit.test("Toggle display of original/translated version of chatter message", async (assert) => {
+    const pyEnv = await startServer();
+    const partnerId = pyEnv["res.partner"].create({});
+    pyEnv["mail.message"].create({
+        model: "res.partner",
+        body: "Al mal tiempo, buena cara.",
+        author_id: pyEnv.odoobotId,
+        res_id: partnerId,
+    });
+    const { openFormView } = await start({
+        mockRPC(route) {
+            if (route === "/discuss/message/translate") {
+                assert.step("Request");
+                return { value: "To bad weather, good face.", source: "Spanish", error: null };
+            }
+        },
+    });
+    await openFormView("res.partner", partnerId);
+    await click("button[title='Expand']");
+    await contains("span[title='Translate']");
+    await contains("span[title='Revert']", { count: 0 });
+    // Click acts as a toogle affecting its appearence and the actual message content displayed.
+    await click("span[title='Translate']");
+    await click("button[title='Expand']");
+    await contains(
+        `.o-mail-Message:contains("To bad weather, good face. (Translated from: Spanish)")`
+    );
+    await contains("span[title='Translate']", { count: 0 });
+    await contains("span[title='Revert']");
+    await click("span[title='Revert']");
+    await click("button[title='Expand']");
+    await contains(`.o-mail-Message:contains("Al mal tiempo, buena cara.")`);
```

Comment:
Note that jquery `contains:` selector is on its way to being removed; use the `text` option instead:
```suggestion
    await contains(".o-mail-Message", { text: "Al mal tiempo, buena cara" });
```